x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/EvaluatorImplementer.java

```diff
@@ -114,20 +131,30 @@ private MethodSpec ctor() {
         return builder.build();
     }
 
+    @SuppressWarnings("checkstyle:DescendantToken")
     private MethodSpec eval() {
         MethodSpec.Builder builder = MethodSpec.methodBuilder("eval").addAnnotation(Override.class);
         builder.addModifiers(Modifier.PUBLIC).returns(BLOCK).addParameter(PAGE, "page");
-
+        if (mv) {
+            // The first two parameters(builder, position) are place holders, no need to resolve them
+            processFunction.args.remove(0);
+            processFunction.args.remove(0);
+        }
         processFunction.args.stream().forEach(a -> a.evalToBlock(builder));
         String invokeBlockEval = invokeRealEval(true);
-        processFunction.args.stream().forEach(a -> a.resolveVectors(builder, invokeBlockEval));
-        builder.addStatement(invokeRealEval(false));
+        if (mv) builder.addStatement(invokeBlockEval);
+        else {
+            processFunction.args.stream().forEach(a -> a.resolveVectors(builder, invokeBlockEval));
+            builder.addStatement(invokeRealEval(false));
+        }
         processFunction.args.stream().forEach(a -> a.closeEvalToBlock(builder));
         return builder.build();
     }
 
     private String invokeRealEval(boolean blockStyle) {
-        StringBuilder builder = new StringBuilder("return eval(page.getPositionCount()");
+        StringBuilder builder;
+        if (mv) builder = new StringBuilder("return evalMv(page.getPositionCount()");
```

Comment:
Ah! This is what checkstyle is about - it wants
```
if (mv) {
  builder = 
} else {
  builder =
}
```

It's long, but it's easier to read. You can scan the file and *know* there's an `if` statement there - you can't with this format. Java folks are so used to this format I actually couldn't tell you if that `;` on the end of this line is correct. I just never write this way.